backend/graph/api/model/chapter.go

```diff
@@ -0,0 +1,67 @@
+package model
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/bcc-code/bcc-media-platform/backend/common"
+	"github.com/bcc-code/bcc-media-platform/backend/user"
+	"github.com/bcc-code/bcc-media-platform/backend/utils"
+)
+
+func ChapterFrom(ctx context.Context, tm *common.TimedMetadata, loaders *common.BatchLoaders) *Chapter {
+	ginCtx, _ := utils.GinCtx(ctx)
+	languages := user.GetLanguagesFromCtx(ginCtx)
+	title := tm.Title.Get(languages)
+	phrase, _ := loaders.PhraseLoader.Get(ctx, tm.ChapterType.Value)
+	emptyTitle := title == ""
+	if emptyTitle && phrase != nil {
+		title = phrase.Value.Get(languages)
+	}
+
+	switch tm.ChapterType {
+	case common.ChapterTypeSong, common.ChapterTypeSingAlong:
+		if !tm.SongID.Valid {
+			break
+		}
+		song, _ := loaders.SongLoader.Get(ctx, tm.SongID.UUID)
```

Comment:
This seems a bit suspicious in the context. In my head the `XFrom` are only data wrangling functions that don't have side effects. Something to look at but don't need to change now.

Reply:
i agree 500%